src/chains/algorand.ts

```diff
@@ -0,0 +1,74 @@
+const algosdk = require('algosdk');
+const nacl = require('tweetnacl');
+
+function privateKeyToArray(privateKey: string) {
+    return Buffer.from(privateKey.substring(2,privateKey.length), 'hex').toJSON().data
+}
+
+function keyPairFromSecretKey(sk: Uint8Array) {
+    return nacl.sign.keyPair.fromSecretKey(sk);
+  }
+
+export default class utils {
+
+    static createWallet(): object {
+        const wallet = algosdk.generateAccount();
+        const mnemonic = algosdk.secretKeyToMnemonic(wallet.sk);
+        const privateKey = '0x' + Buffer.from(wallet.sk).toString('hex')
+        const keypair = keyPairFromSecretKey(wallet.sk)
+        const publicKey = '0x' + Buffer.from(keypair.publicKey).toString('hex')
+
+        return {
+            mnemonic,
+            privateKey,
+            publicKey,
+            address: wallet.addr
+        }
+    }
+
+    static getWallet(mnemonic: string): object {
+        const wallet = algosdk.mnemonicToSecretKey(mnemonic)
+        const privateKey = '0x' + Buffer.from(wallet.sk).toString('hex')
+        const keypair = keyPairFromSecretKey(wallet.sk)
+        const publicKey = '0x' + Buffer.from(keypair.publicKey).toString('hex')
+
+        return {
+            mnemonic: mnemonic,
+            privateKey,
+            publicKey,
+            address: wallet.addr
+        }
+    }
+
+    static getPublicKey() {
+        throw new Error('Not implemented');
```

Comment:
In ETH the address is the public key... is that the same in ALGO?

Reply:
Yes, that's true for ALGO as well.